packages/vm/lib/evm/opcodes/functions.ts

```diff
@@ -469,15 +469,18 @@ export const handlers: Map<number, OpHandler> = new Map([
       const [memOffset, dataOffset, dataLength] = runState.stack.popN(3)
 
       subMemUsage(runState, memOffset, dataLength)
-      runState.eei.useGas(
-        new BN(runState._common.param('gasPrices', 'copy')).imul(divCeil(dataLength, new BN(32)))
-      )
 
-      const data = getDataSlice(runState.eei.getCallData(), dataOffset, dataLength)
-      const memOffsetNum = memOffset.toNumber()
-      const dataLengthNum = dataLength.toNumber()
-      runState.memory.extend(memOffsetNum, dataLengthNum)
-      runState.memory.write(memOffsetNum, dataLengthNum, data)
+      if (!dataLength.eqn(0)) {
+        runState.eei.useGas(
+          new BN(runState._common.param('gasPrices', 'copy')).imul(divCeil(dataLength, new BN(32)))
+        )
```

Comment:
Is this correct that these `useGas()` calls are within the conditions? If I read the code correct, this still results in a > 0 gas usage value also when `dataLength` === 0?

Reply:
If `dataLength` is 0, then you multiply by `0` 😄

Reply:
Ah, sorry, I misread thinking that `divCeil(dataLength, new BN(32))` would just return the ceil from both values.